sql/src/main/java/org/apache/druid/sql/guice/SqlModule.java

```diff
@@ -63,25 +55,21 @@ public void configure(Binder binder)
     if (isEnabled()) {
       Calcites.setSystemProperties();
 
-      JsonConfigProvider.bind(binder, "druid.sql.planner", PlannerConfig.class);
-      JsonConfigProvider.bind(binder, "druid.sql.avatica", AvaticaServerConfig.class);
-      LifecycleModule.register(binder, DruidSchema.class);
       binder.bind(ViewManager.class).to(NoopViewManager.class).in(LazySingleton.class);
 
-      // Add empty SqlAggregator binder.
-      Multibinder.newSetBinder(binder, SqlAggregator.class);
+      binder.install(new DruidCalciteSchemaModule());
```

Comment:
I'm wondering why you wanted to split these up? It seemed nice to me to have all the core SQL-related bindings in one file.

Reply:
I usually like smaller modules because it's easier to rationalize what needs to be installed in that module. Every package has a module that describes how implementations in it's package are provided to the rest of the code. The other advantage of this is you can keep all the implementations package private and only expose interfaces to other packages - which means there's less accidental coupling between packages.

This also makes testing each module easier (I realize I haven't written tests for them yet, but they're coming)

Reply:
The only reason I asked was because I was concerned that having things be too split up would make it hard for people to guess what functionality should be in which file (at least with something named SqlModule it's clear that all the SQL stuff goes in there).

The easier testability is nice though, so you convinced me it's worth it. Let's just make sure it's as clear as possible from the namings what stuff goes in what module (or at least from the javadocs).

Reply:
A pattern I've used in the past is 1 module per sub package. The source directory structure should already be broken up so that related classes are under the same package.